packages/firestore/test/unit/local/query_cache.test.ts

```diff
@@ -47,6 +48,17 @@ describe('MemoryQueryCache', () => {
   });
 
   genericQueryCacheTests();
+
+  it('cannot remove nonexistent query', async () => {
+    // We don't check for existence in the indexeddb query cache, but we do for
+    // memory. So, this test is specific to the memory-backed query cache.
+    try {
+      await cache.removeQueryData(testQueryData(QUERY_ROOMS, 1, 1));
+      fail('should not have been able to remove nonexistent query');
```

Comment:
This `fail` method just throws an exception, which you'll then swallow...  and so I think this test can't fail.  You probably want `expect.fail()` instead, which hooks into the test runner to register the failure rather than relying on an exception.

You could also consider writing it without await, since .then() lets you attach both a success and rejection handler at once:

```
return cache.removeQueryData(...).then(
  () => fail(' ... '),
  (err) => { /* expected. */ });
```

Reply:
Alternatively I don't feel strongly that we need tests to cover assertions and since you didn't add tests for updateQueryData() or addQueryData(), I'd be somewhat inclined to just drop this test.  If you do that, please undo the QUERY_* / testQueryData() refactor.

Reply:
Sounds good.